test/token/WstEthL2Token.t.sol

```diff
@@ -0,0 +1,285 @@
+
+        // Perform the upgrade
+        vm.startPrank(governance);
+        OzUpgrades.upgradeProxy(
+            address(token),
+            "WstEthL2TokenV2Fake.sol",
+            abi.encodeCall(WstEthL2TokenV2Fake.initializeFakeV2, (_initialValue)),
+            _upgradeProxyOpts()
+        );
+        vm.stopPrank();
+
+        // Ensure the contract is initialized correctly
+        WstEthL2TokenV2Fake _tokenV2 = WstEthL2TokenV2Fake(address(token));
+        assertEq(_tokenV2.fakeStateVar(), _initialValue);
+
+        // Ensure we can exercise pre-upgrade functionality, such as minting
+        // Ensure the storage applied pre-upgrade, in this case the minter, still functions as expected
+        vm.prank(_minter);
+        _tokenV2.mint(_minter, _mintAmount);
+        assertEq(_tokenV2.balanceOf(_minter), _mintAmount);
+
+        // Ensure we can exercise some new functionality included in the upgrade
+        vm.prank(_minter);
+        vm.expectEmit();
+        emit WstEthL2TokenV2Fake.FakeStateVarSet(_initialValue, _nextValue);
+        _tokenV2.setFakeStateVar(_nextValue);
+        assertEq(_tokenV2.fakeStateVar(), _nextValue);
+
+        // Ensure the role ACL applied to the new method works
+        address _notMinter = address(uint160(uint256(keccak256(abi.encode(_minter)))));
+        vm.expectRevert(
+            abi.encodeWithSelector(WstEthL2Token.UnauthorizedAccount.selector, _notMinter)
+        );
+        vm.prank(_notMinter);
+        token.mint(_notMinter, _mintAmount);
+
+        // Ensure initialization cannot be called again
+        vm.expectRevert(Initializable.InvalidInitialization.selector);
+        _tokenV2.initialize("Wrapped Staked Eth", "wstEth", governance);
+        vm.expectRevert(Initializable.InvalidInitialization.selector);
+        _tokenV2.initializeFakeV2(_nextValue);
+    }
+}
+
+contract Mint is WstEthL2TokenTest {
+    function testFuzz_CorrectlyMintTokensForANewAddress(address _account, uint256 _amount) public {
+        _amount = bound(_amount, 0, MAX_INT);
+        vm.assume(_account != address(0));
+        vm.prank(minter);
+        token.mint(_account, _amount);
+
+        assertEq(token.balanceOf(_account), _amount);
+    }
+
+    function testFuzz_RevertIf_CalledByNonMinterAddress(
+        address _caller,
+        uint256 _mintAmount
+    ) public {
+        vm.assume(_caller != minter);
+        _mintAmount = bound(_mintAmount, 0, MAX_INT);
+
+        vm.expectRevert(abi.encodeWithSelector(WstEthL2Token.UnauthorizedAccount.selector, _caller));
+        vm.prank(_caller);
+        token.mint(_caller, _mintAmount);
+    }
+}
+
+contract Burn is WstEthL2TokenTest {
+    function testFuzz_CorrectlyBurnTokensForANewAddress(
+        uint256 _mintAmount,
+        uint256 _burnAmount
+    ) public {
+        _mintAmount = bound(_mintAmount, 0, MAX_INT);
+        _burnAmount = bound(_burnAmount, 0, MAX_INT);
+        vm.assume(_mintAmount >= _burnAmount);
```

Comment:
Could we not just bound the burnAmount to the mintAmount instead of the extra assume?